include/stl2/detail/iterator/basic_iterator.hpp

```diff
@@ -722,7 +722,7 @@ STL2_OPEN_NAMESPACE {
     {
       auto tmp = *this;
       --*this;
-      return *this;
+      return tmp;
```

Comment:
> Or simply `return prev(*this);`

We want the "old" value which is `next(*this)`, not `prev(*this)`. `--*this; return next(*this);` would be semantically equivalent, but I conjecture that `auto tmp = *this; --*this; return tmp;` is cheaper in general. A copy must be made in any case, copying the original value will be cheaper if the optimizer can't determine that `++` simply reverses the effect of `--`.